module/VuFind/src/VuFind/Db/Table/OaiResumption.php

```diff
@@ -85,11 +86,51 @@ public function removeExpired()
      *
      * @param string $token The resumption token to retrieve.
      *
-     * @return ?\VuFind\Db\Row\OaiResumption
+     * @return     ?\VuFind\Db\Row\OaiResumption
+     * @deprecated Use OaiResumption::findWithId
      */
     public function findToken($token)
     {
-        return $this->select(['id' => $token])->current();
+        return $this->findWithId($token);
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key; return null if it
+     * is not found.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return ?\VuFind\Db\Row\OaiResumption
+     */
+    public function findWithId(string $id): ?OaiResumptionEntityInterface
+    {
+        return $this->select(['id' => $id])->current();
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key and where the token is null.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return     ?\VuFind\Db\Row\OaiResumption
+     * @deprecated This function is for BC and can be removed in the future.
+     */
+    final public function findWithIdBc(string $id): ?OaiResumptionEntityInterface
```

Comment:
If you rename the service method, maybe this should match, i.e. `findWithLegacyIdToken`... and again, maybe the deprecation should be a todo for the moment.